### DISCUSSION TITLE
Multi Button press Delay

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#19400] [Q&A]
Continuing with Issue #[10509](https://github.com/arendst/Tasmota/issues/10509) **[Feature Request] Multi Button press Delay**

Suggestion was to:

[stefanbode](https://github.com/stefanbode) commented [Jan 20, 2021](https://github.com/arendst/Tasmota/issues/10509#issuecomment-763445044)

```
You can try to change the code directly to check if this helps your use case. You will find the responsible line in support_button.ino line: 243

        Button.window_timer[button_index] = loops_per_second / 2;  // 0.5 second multi press window

Remove the /2 and you have 1 second to press again instead of 0.5 seconds.
```

However when I looked at 12.3.1.3 code I see two files V2 and V3.
```
-rw-r--r-- 1 ubuntu ubuntu  19855 Jan  5  2023 support_button_v2.ino
-rw-r--r-- 1 ubuntu ubuntu  25133 Jan  5  2023 support_button_v3.ino

```
Which one should I modify?

Both files have the same code in corresponding lines:

![image](https://github.com/arendst/Tasmota/assets/84637540/21e2bd31-240e-499b-84b0-526f393eb3ee)

![image](https://github.com/arendst/Tasmota/assets/84637540/63e5d24b-c8d3-4e0d-9b97-bbe3359cc296)

The latest version of Tasmota has only one file V4. 
![image](https://github.com/arendst/Tasmota/assets/84637540/cbcc89a6-e79b-4105-a308-c965afe79f65)

However I prefer to test it with 12.3.1.3 as most of my devices are working with this version and haven't given a thought of upgrading all of them yet.

**Comment by stefanbode** [comment]
Yes you should go with version v4. As you can see there was no change during the last year. As this is a quite rare request I assume the only change we can offer is a compiler option to change this. Let me know if this helps and I can submit a small PR. Therefore you do not maintain code but compile your own option.

**Reply by jkwim** [reply]
Yes, a compiler option would have been more than sufficient. Thanks.

**Comment by stefanbode** [comment]
Submitted a pull request: https://github.com/arendst/Tasmota/pull/19428
